luigi/target.py

```diff
@@ -348,3 +349,23 @@ def __exit__(self, exc_type, exc, traceback):
         if exc_type:
             return
         return super(AtomicLocalFile, self).__exit__(exc_type, exc, traceback)
+
+
+class LazyFormatMixin:
+    _format = None
```

Comment:
Be vary of class variables vs instance variables when it comes to modifying them

Reply:
You want this changed to 1. set _format lazily or 2. require subclasses to populate `_format = None`? Normally only do this when the default is None since you can't modify the class instance via self.

Reply:
I guess i was thinking

```python
class MyClass:
    _var = None

# vs

class YourClass:
    def __init__(self, *args, **kwargs):
        super(YourClass, self).__init__(*args, **kwargs)
        self._var = None
```